internal/system/agent/services.go

```diff
@@ -51,114 +53,208 @@ func InvokeOperation(action string, services []string) bool {
 
-	configuration := make(map[string]interface{})
+	c := ConfigRespMap{}
+	c.Configuration = map[string]interface{}{}
 
-	// Loop through requested actions, along with respectively-supplied parameters.
+	// Loop through requested actions, along with (any) respectively-supplied parameters.
 	for _, service := range services {
 
+		// Check whether SMA does _not_ know of ServiceKey ("service") as being one for one of its ready-made list of clients.
 		if !isKnownServiceKey(service) {
-			logs.LoggingClient.Warn("unknown service found getting configuration", "service name", service)
-		}
+			LoggingClient.Info(fmt.Sprintf("service %s not known to SMA as being in the ready-made list of clients", service))
+
+			// Service unknown to SMA, so ask the Registry whether `service` is available.
+			serviceError := registryClient.IsServiceAvailable(service)
+			if serviceError != nil {
+				msg := fmt.Sprintf("Registry responded with %s service not available, with error code: %s", service, serviceError.Error())
+				LoggingClient.Error(msg)
+				c.Configuration[service] = fmt.Sprintf(msg)
+			} else {
+				LoggingClient.Info(fmt.Sprintf("Registry responded with %s service available", service))
 
-		responseJSON, err := clients[service].FetchConfiguration(ctx)
-		if err != nil {
-			configuration[service] = fmt.Sprintf("%s get config error: %s", service, err.Error())
-			logs.LoggingClient.Error("error retrieving configuration", "service name", service, "errMsg", err.Error())
+				// Since service is unknown to SMA, ask the Registry for a ServiceEndpoint associated with `service`
+				e, err := registryClient.GetServiceEndpoint(service)
+				if err != nil {
+					msg := fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err)
+					LoggingClient.Error(msg)
+					return c, err
+				} else {
+					// Preparing to add the specified key to the map where the value will be the respective GeneralClient
+					clientInfo.Protocol = Configuration.Service.Protocol
+					clientInfo.Host = e.Host
+					clientInfo.Port = e.Port
+
+					Configuration.Clients[e.ServiceId] = clientInfo
+					params := types.EndpointParams{
+						ServiceKey:  e.ServiceId,
+						Path:        "/",
+						UseRegistry: usingRegistry,
+						Url:         Configuration.Clients[e.ServiceId].Url() + clients.ApiConfigRoute,
+						Interval:    internal.ClientMonitorDefault,
+					}
+					// TODO: The following note is related to future work:
+					// TODO: With the current deployment strategy, GeneralClient's func init(params types.EndpointParams) {...} returns blank "url"...
+					// TODO: [Need a manifest-like functionality in future...]
+					// Add the service key to the map where the value is the respective GeneralClient
+					generalClients[e.ServiceId] = general.NewGeneralClient(params, startup.Endpoint{RegistryClient: &registryClient})
+
+					c.Configuration[service] = ""
+
+					responseJSON, em := generalClients[e.ServiceId].FetchConfiguration(ctx)
```

Comment:
What is `em`? The convention for errors is `err`.

Reply:
Yep, I had renamed it only so as to avoid the `err` variable getting shadowed. I will relook this one...